app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsViewModel.kt

```diff
@@ -133,60 +69,140 @@ class SessionDetailsViewModel @JvmOverloads constructor(
     val subtitleFont = Font.Roboto.Light
     val titleFont = Font.Roboto.BoldCondensed
 
-    fun onOptionsMenuItemSelected(menuItemId: Int) = when (menuItemId) {
-        R.id.menu_item_feedback -> {
-            val uri = feedbackUrl.toUri()
-            viewActionHandler.openFeedback(uri)
-            true
-        }
-        R.id.menu_item_share_session,
-        R.id.menu_item_share_session_text -> {
-            val formattedSession = session.toPlainText(timeZoneId)
-            viewActionHandler.shareAsPlainText(formattedSession)
-            true
+    val selectedSessionParameter: LiveData<SelectedSessionParameter> = repository.selectedSession
+        .map { it.toSelectedSessionParameter() }
+        .asLiveData(executionContext.database)
+
+    val openFeedBack = SingleLiveEvent<Uri>()
+    val shareSimple = SingleLiveEvent<String>()
+    val shareJson = SingleLiveEvent<String>()
+    val addToCalendar = SingleLiveEvent<Session>()
+    val setAlarm = SingleLiveEvent<Unit>()
+    val navigateToRoom = SingleLiveEvent<Uri>()
+    val closeDetails = SingleLiveEvent<Unit>()
+
+    private fun Session.toSelectedSessionParameter(): SelectedSessionParameter {
+        val useDeviceTimeZone = repository.readUseDeviceTimeZoneEnabled()
+        val formattedZonedDateTime = formattingDelegate.getFormattedDateTimeShort(useDeviceTimeZone, dateUTC, timeZoneOffset)
+        val formattedAbstract = markdownConversion.markdownLinksToHtmlLinks(abstractt)
+        val formattedDescription = markdownConversion.markdownLinksToHtmlLinks(description)
+        // language=regex
+        val linksHtml = getLinks().replace("\\),".toRegex(), ")<br>")
+        val formattedLinks = markdownConversion.markdownLinksToHtmlLinks(linksHtml)
+        val sessionUrl = sessionUrlComposition.getSessionUrl(this)
+        val sessionLink = if (sessionUrl.isEmpty()) "" else "<a href=\"$sessionUrl\">$sessionUrl</a>"
```

Comment:
Same with `linksHtml`

Reply:
:heavy_check_mark: Added in b454420c6317914f52318589c67daf1fd03dffe4.